exchange/index.go

```diff
@@ -453,6 +476,145 @@ func (idx *Index) evict(size uint64) uint64 {
 	return evicted
 }
 
+func (idx *Index) tagForEviction(ref *DataRef) error {
+	idx.emu.Lock()
+	defer idx.emu.Unlock()
+
+	return idx.walkDAG(ref.PayloadCID, func(blk blocks.Block) error {
+		idx.evictionSet.Add(blk.Cid())
+		return nil
+	})
+}
+
+func (idx *Index) GC() {
+	idx.emu.Lock()
+	defer idx.emu.Unlock()
+
+	// exit if there is nothing to evict
+	if idx.evictionSet.Len() == 0 {
+		return
+	}
+
+	gcbl, ok := idx.bstore.(blockstore.GCBlockstore)
+	if ok {
```

Comment:
```suggestion
	if !ok {
	           return fmt.Errorf("blockstore is not a GCBlockstore")
```
I think we prob don't want to continue if we can't get a lock